src/main/java/com/ioet/bpm/people/boundaries/PersonController.java

```diff
@@ -51,17 +61,33 @@ public PersonController(PersonRepository personRepository) {
 
     @PutMapping("/{id}")
     public ResponseEntity<Person> updatePerson(@PathVariable(value = "id") String personId,
-                                             @Valid @RequestBody Person personToUpdate) {
+                                               @Valid @RequestBody Person personToUpdate) {
 
         Optional<Person> personFound = personRepository.findById(personId);
-
         if (personFound.isPresent()) {
-            personToUpdate.setId(personId);
+            personToUpdate.setId(personFound.get().getId());
+            personToUpdate.setPassword(personFound.get().getPassword());
             Person updatedPerson = personRepository.save(personToUpdate);
             return new ResponseEntity<>(updatedPerson, HttpStatus.OK);
         }
         return new ResponseEntity<>(HttpStatus.NOT_FOUND);
+    }
+
+    @PostMapping("/change-password/{id}")
+    public ResponseEntity<Person> changePassword(@PathVariable(value = "id") String personId,
+                                                 @Valid @RequestBody Person personToUpdate) {
 
+        Optional<Person> personFound = personRepository.findById(personId);
+        if (personFound.isPresent()) {
+            personToUpdate.setId(personFound.get().getId());
+            personToUpdate.setPassword(passwordManagementService.generatePassword(personToUpdate.getPassword()));
+            Person updatedPerson = personRepository.save(personToUpdate);
+
+            PasswordHistory passwordHistory = passwordManagementService.createPasswordHistory(personToUpdate);
+            passwordHistoryRepository.save(passwordHistory);
 
```

Comment:
I think these all lines should be move to passwordManagementService

```
PasswordManagementService#changePassword(personId, newPassword){
   Person person = personRepository.findById(personId).get();
   person.setPassword(this.generatePassword(newPassword));
   recordPasswordHistory(person);   
}
```
and then:
```
 private recordPasswordHistory(person){
    PasswordHistory passwordHistory = createPasswordHistory(person);
    passwordHistoryRepository.save(passwordHistory);
 }
```